core/src/poh_service.rs

```diff
@@ -33,6 +34,9 @@ impl PohService {
                 if poh_config.hashes_per_tick.is_none() {
                     Self::sleepy_tick_producer(poh_recorder, &poh_config, &poh_exit_);
                 } else {
+                    if let Some(cores) = core_affinity::get_core_ids() {
```

Comment:
@pgarg66 It's important that you comment the hell out of side-effects like this that can't be unit-tested. What would discourage someone from deleting this code?

Reply:
@garious , sure, I'll add some comments in the next PoH related PR (coming soon)

Reply:
@garious this PR has the comment: https://github.com/solana-labs/solana/pull/4396